src/ReadingBuddy.java

```diff
@@ -925,15 +1034,35 @@ public static void manageCurrentBooks(){
                             currentUser.getUsername());
                             currentUser.addToReadBooks(toAdd);
                             allReadBooks.add(toAdd);
-                        } else {
-                            System.out.println("\n" + selectedBook.getTitle() + " is already in your Read Books\n");
                         }
-                        removeCurrentBook((numSelected-1), selectedBook);
-                        //later will need to add more to preserve and update group relationship
+                        //NEW SPRINT3: only remove book if NOT in book group
+                        if(selectedBook.getGroup() < 0){
+                            removeCurrentBook(selectedBook);
+                        }else{
+                            //if in a book group just set+update isDone attribute
+                            selectedBook.setIsDone(true);
+                            updateCurrentlyReading();
+                            //!!!FIX: update tracking of doneness in book group
```

Comment:
i forgot to get rid of that note to myself, but can be deleted. the code below the comment updates the user's isDone attribute in the group to true